hera_filters/dspec.py

```diff
@@ -2960,6 +2978,36 @@ def sparse_linear_fit_2D(
         axis_1_basis.shape[-1] * axis_2_basis.shape[-1],  # i * j
     )
 
+    if precondition_solver:
+        # Compute separate preconditioners for the two axes
+        # Start by computing separable weights for the two axes
+        u, s, v = np.linalg.svd(weights, full_matrices=False)
```

Comment:
Try https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.linalg.svds.html

Reply:
Using this now